dotnet/Simple.Sauce/SauceOptions.cs

```diff
@@ -36,10 +57,88 @@ public SauceOptions()
+                }
+                else if (GetEnvironmentVariable(knownCITools["Bamboo"]) != null)
+                {
+                    return GetEnvironmentVariable("bamboo_shortJobName") + ": " + GetEnvironmentVariable("bamboo_buildNumber");
+                }
+                else if (GetEnvironmentVariable(knownCITools["Travis"]) != null)
+                {
+                    return GetEnvironmentVariable("TRAVIS_JOB_NAME") + ": " + GetEnvironmentVariable("TRAVIS_JOB_NUMBER");
+                }
+                else if (GetEnvironmentVariable(knownCITools["Circle"]) != null)
+                {
+                    return GetEnvironmentVariable("CIRCLE_JOB") + ": " + GetEnvironmentVariable("CIRCLE_BUILD_NUM");
+                }
+                else if (GetEnvironmentVariable(knownCITools["GitLab"]) != null)
+                {
+                    return GetEnvironmentVariable("CI_JOB_NAME") + ": " + GetEnvironmentVariable("CI_JOB_ID");
+                }
+                else if (GetEnvironmentVariable(knownCITools["TeamCity"]) != null)
+                {
+                    return GetEnvironmentVariable("TEAMCITY_PROJECT_NAME") + ": " + GetEnvironmentVariable("BUILD_NUMBER");
+                }
+                else
+                {
+                    return "Build Time: " + DateTime.Now;
+                }
+            }
+            set { _buildName = value; }
+        }
+
+        internal DriverOptions ToDriverOptions()
+        {
+            var w3cCapabilities = SeleniumOptions;
+            //TODO temporary solution to get the code working
+            var sauceConfiguration = new Dictionary<string, object>
+            {
+                ["username"] = Environment.GetEnvironmentVariable("SAUCE_USERNAME"),
+                ["accessKey"] = Environment.GetEnvironmentVariable("SAUCE_ACCESS_KEY")
+            };
+            //TODO add if logic from the toCapabilities() in Java
+
+            if (BrowserName == Browser.Edge)
+                w3cCapabilities = new EdgeOptions();
+
+            w3cOptions.ForEach(capability => AppendCapabilityIfDefined(sauceConfiguration, capability));
+
+            w3cCapabilities.AddAdditionalOption("sauce:options", sauceConfiguration);
+            return w3cCapabilities;
+        }
+
+        private void AppendCapabilityIfDefined(Dictionary<string, object> sauceConfiguration, string capability)
+        {
+            var capabilityValue = TryToGetCapabilityValue(capability);
+            if (capabilityValue != null)
+                sauceConfiguration.Add(capability, capabilityValue);
+        }
+
+        private object TryToGetCapabilityValue(string capability)
+        {
+            try
+            {
+                return GetType().GetProperty(capability).GetValue(this);
+            }
+            catch (NullReferenceException)
+            {
+                return null;
+            }
+            //PropertyInfo pinfo = typeof(SauceOptions).GetProperty(capability);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/SauceLabs/simple_sauce/pullRequest?prid=4945851)